src/App.tsx

```diff
@@ -3,24 +3,50 @@ import { Navbar } from 'reactstrap';
 
 import { Footer } from './components/Footer';
 import { WebMapContainer } from './containers/WebMapContainer';
+import { IdentityContainer, IdentityManagerEvent } from './containers/IdentityContainer';
 
 import './App.css';
 
-class App extends React.Component<object, object> {
+const portalUrl = 'https://www.arcgis.com/sharing';
+const appId = 'y4Lx1l6456Mbf85z';
+
+interface AppProps {}
+interface AppState {
+  credential?: __esri.Credential;
+}
+
+class App extends React.Component<AppProps, AppState> {
+  constructor(props: AppProps) {
+    super(props);
+    this.state = {};
+    this.handleLoginChange = this.handleLoginChange.bind(this);
+  }
+
+  handleLoginChange(event: IdentityManagerEvent) {
```

Comment:
One last nitpick: this component shouldn't need to adapt to the conventions of its children, the children should adapt to the parent. This will make it easier to pass this method to other child components if need be.